sds_in_a_box/SDSCode/cognito_signup_message.py

```diff
@@ -0,0 +1,16 @@
+import os
+import boto3 
+
+def lambda_handler(event, context):
+    client = boto3.client('cognito-idp')
+    domain_desc = client.describe_user_pool_domain(Domain=os.environ["COGNITO_DOMAIN_PREFIX"])
+    clients = client.list_user_pool_clients(UserPoolId = domain_desc["DomainDescription"]['UserPoolId'])
+    command_line_client = clients['UserPoolClients'][0]['ClientId']
+    
+    if 'triggerSource' in event:
```

Comment:
What event is triggering this lambda?

Reply:
Whenever a user receives an email from the cognito userpool, first it will send the event through this lambda.  This lambda's job is to take that event, change it in some way, and then return the changed event back to cognito.  

So in this lambda's case, we only want to change the message a user receives when their account is first created by the admin (i.e from the event "CustomMessage_AdminCreateUser").  What I wanted to do here was simply add a link to the signup URL, because by default congito doesn't actually tell the user where to go to verify their account and change their initial password.  

There are several other events that will cause this lambda to trigger.  For example if the user wants to reset their password, cognito sends them an email which triggers this lambda (with the tigger "CustomMessage_ForgotPassword").  However, we don't want to change those specific emails from the user, so the event in that case is just passed along untouched.  

To make this all more clear, I'll add some documentation to the top of this lambda

Reply:
thanks!